app/connectors/propertyLinking/PropertyLinkConnector.scala

```diff
@@ -103,29 +103,27 @@ class PropertyLinkConnector @Inject()(config: ServicesConfig, http: HttpClient)(
         pagination: PaginationParams,
         organisationId: Long,
         agentOrganisationId: Long,
-        agentAppointed: Option[String] = None
+        agentAppointed: Option[String] = None,
+        agentCode: Long
   )(implicit hc: HeaderCarrier): Future[OwnerAuthResult] = {
-
-    val ownerAuthResult = http.GET[OwnerAuthResult](
-      s"$baseUrl/owner/property-links/appointable",
-      List(
-        searchParams.address.map("address"  -> _),
-        searchParams.baref.map("baref"      -> _),
-        searchParams.agent.map("agent"      -> _),
-        searchParams.status.map("status"    -> _),
-        Some("sortfield"                    -> searchParams.sortfield.toString),
-        Some("sortorder"                    -> searchParams.sortorder.toString),
-        agentAppointed.map("agentAppointed" -> _),
-        Some("organisationId"               -> organisationId.toString),
-        Some("agentOrganisationId"          -> agentOrganisationId.toString)
-      ).flatten ++
-        List(
-          "startPoint"           -> pagination.startPoint.toString,
-          "pageSize"             -> pagination.pageSize.toString,
-          "requestTotalRowCount" -> pagination.requestTotalRowCount.toString
+    val ownerAuthResult = agentAppointed.getOrElse("BOTH") match {
+      case "NO" =>
+        getMyOrganisationsPropertyLinks(searchParams, pagination)
+      case _ =>
+        http.GET[OwnerAuthResult](
+          s"$baseUrl/my-organisation/agents/$agentCode/available-property-links",
+          List(
```

Comment:
You're wrapping half of the mandatory parameters in Some. Then flattening. Either do all of them in Some and flatten at the end, or - since it's only the address that is optional - so you could this instead:
```scala
searchParams.address.map("address" -> _).toList ++ List(
 "sortField"            -> searchParams.sortfield.toString,
 "sortOrder"            -> searchParams.sortorder.toString,
 "startPoint"           -> pagination.startPoint.toString,
 "pageSize"             -> pagination.pageSize.toString,
 "requestTotalRowCount" -> pagination.requestTotalRowCount.toString
)
```